### DISCUSSION TITLE
Nextjs + React 19 - title and meta components

### DISCUSSION THREAD

**Discussion by OllieJennings** [vercel/next.js#62581] [App Router]
Hi,

Not sure if this has been asked yet or not (couldn't find anything)

With the upcoming release of React 19 later this year: https://react.dev/blog/2024/02/15/react-labs-what-we-have-been-working-on-february-2024.

The new `title` and `meta` components are now in canary, how is NextJS and its `generateMetadata` functionality going to evolve?

For me, `generateMetadata` never worked as we use GraphQL and do not want to be firing more than one query per page, and therefore want to set the metadata from within the component.

**Comment by icyJoseph** [comment]
I have done a bit of code exploration, and figured out that Next.js uses these at the end of day. See more here, https://github.com/vercel/next.js/discussions/75054#discussioncomment-11872343
